repos/fdbt-site/src/pages/defineCapPricingPerDistance.tsx

```diff
@@ -0,0 +1,144 @@
+                                        setNumberOfProducts(numberOfCap + 1);
+                                        setCapPricingPerDistanceData({
+                                            ...capPricingPerDistanceData,
+                                            [`distanceTo${numberOfCap - 1}`]: '',
+                                            [`distanceTo${numberOfCap}`]: 'Max',
+                                        });
+                                    }}
+                                >
+                                    Add another product
+                                </button>
+                            ) : (
+                                ''
+                            )}
+
+                            {numberOfCap > 1 ? (
+                                <button
+                                    id="remove-button"
+                                    type="button"
+                                    className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3"
+                                    onClick={(): void => {
+                                        setCapPricingPerDistanceData((current) => {
+                                            const copy = { ...current };
+                                            delete copy[`distanceTo${numberOfCap - 1}`];
+                                            delete copy[`distanceFrom${numberOfCap - 1}`];
+                                            delete copy[`pricePerKm${numberOfCap - 1}`];
+                                            return { ...copy, [`distanceTo${numberOfCap - 2}`]: 'Max' };
+                                        });
+                                        setNumberOfProducts(numberOfCap - 1);
+                                    }}
+                                >
+                                    Remove last product
+                                </button>
+                            ) : (
+                                ''
+                            )}
+                        </div>
+                        <input
+                            type="submit"
+                            value="Continue"
+                            id="continue-button"
+                            className="govuk-button govuk-!-margin-left-3"
+                        />
+                    </div>
+                </>
+            </CsrfForm>
+        </FullColumnLayout>
+    );
+};
+
+export const getServerSideProps = (ctx: NextPageContextWithSession): { props: DefineCapPricingPerDistanceProps } => {
+    const csrfToken = getCsrfToken(ctx);
+    const capPricePerDistances: CapPricePerDistances[] = getSessionAttribute(
+        ctx.req,
+        CAP_PRICING_PER_DISTANCE_ATTRIBUTE,
+    )?.capPricePerDistances as CapPricePerDistances[];
+    const finalCapPricePerDistances: CapPricingPerDistanceData = { distanceFrom0: '0' };
+    let numberOfCap = 0;
+
+    if (capPricePerDistances) {
```

Comment:
We shouldnt be doing any data building on the serverSideProps, it should all be done inside the API.